src/Components/AnimeShelf.js

```diff
@@ -1,151 +1,18 @@
-    );
-    e.preventDefault();
-  };
-
-  const onClickPrevious = (e) => {
-    setStartIndex(Math.max(startIndex - itemsPerPage, 0));
-    e.preventDefault();
-  };
-
-  const handlers = useSwipeable({
-    onSwiped: (eventData) => console.log("user swiped!", eventData),
-    onSwipedLeft: (eventData) =>
-      setStartIndex(
-        Math.min(startIndex + itemsPerPage, items.length - itemsPerPage)
-      ),
-    onSwipedRight: (eventData) =>
-      setStartIndex(Math.max(startIndex - itemsPerPage, 0)),
-  });
-
-  return (
-    <Box
-      {...handlers}
-      sx={{
-        display: "flex",
-        flexDirection: "column",
-        textAlign: "start",
-        mb: 1,
-        position: "relative",
-      }}
-    >
-      <Box
-        sx={{
-          paddingLeft: { xs: hasPrevious ? "24px" : 0, lg: 0 },
-          paddingRight: { xs: hasNext ? "24px" : 0, lg: 0 },
-        }}
-      >
-        <Grid container spacing={2} columns={columns}>
-          {showGhosts &&
-            ghosts.map((_, index) => (
-              <Grid
-                item
-                key={index}
-                {...breakpoints}
-                sx={{ aspectRatio: "0.7" }}
-              >
-                <Skeleton
-                  variant="rounded"
-                  sx={{ height: "100%", borderRadius: "8px" }}
-                />
-              </Grid>
-            ))}
-          {!showGhosts &&
-            currentItems.map((anime, index) => (
-              <Grid
-                item
-                key={index}
-                {...breakpoints}
-                sx={{ aspectRatio: "0.7", zIndex: selected === index ? 1 : 0 }}
-              >
-                <AnimeCard
-                  anime={anime}
-                  onChangeSelected={(v) => onChangeSelected(index, v)}
-                />
-              </Grid>
-            ))}
-        </Grid>
-      </Box>
-      <Paper
-        elevation={4}
-        sx={{
-          position: "absolute",
-          visibility: hasPrevious ? "visible" : "hidden",
-          left: { xs: 0, lg: "-24px" },
-          top: "calc(50% - 24px)",
-          zIndex: 1,
-          borderRadius: "24px",
-        }}
-      >
-        <IconButton onClick={onClickPrevious} color="inherit">
-          <CaretLeft size={24} />
-        </IconButton>
-      </Paper>
-      <Paper
-        elevation={4}
-        sx={{
-          position: "absolute",
-          visibility: hasNext ? "visible" : "hidden",
-          right: { xs: 0, lg: "-24px" },
-          top: "calc(50% - 24px)",
-          zIndex: 1,
-          borderRadius: "24px",
-        }}
-      >
-        <IconButton onClick={onClickNext} color="inherit">
-          <CaretRight size={24} />
-        </IconButton>
-      </Paper>
-    </Box>
-  );
+  const cannotHover = useMediaQuery("(hover: none)");
+  const coarsePointer = useMediaQuery("(pointer: coarse)");
+
+  const isTouchscreen = cannotHover && coarsePointer;
```

Comment:
Awesome.  Also, thanks for sending attaching that article - I really like this approach of detecting hover-friendly devices.